homeassistant/components/upnp/device.py

```diff
@@ -32,6 +32,16 @@
 )
 
 
+def get_mac_address_from_host(host: str) -> str | None:
+    """Get mac address from host."""
+    ip_addr = ip_address(host)
+    if ip_addr.version == 4:
+        mac_address = get_mac_address(ip=host)
+    else:
+        mac_address = get_mac_address(ip6=host)
+    return mac_address
+
```

Comment:
This does I/O so it needs to run in the executor

Reply:
Done.